modules/migrations/gitea_uploader.go

```diff
@@ -689,6 +689,15 @@ func (g *GiteaLocalUploader) newPullRequest(pr *base.PullRequest) (*models.PullR
 		}
 	} else {
 		head = pr.Head.Ref
+		// Ensure the closed PR SHA still points to an existing ref
+		_, err = git.NewCommand("log", pr.Head.SHA).RunInDir(g.repo.RepoPath())
```

Comment:
`git rev-list -1` would be better. There is almost certainly some code in modules/git that does this for you.

This as is will cause all commits and their summaries to be read in to a string to then immediately be thrown away. Thus this is a hard change request

Reply:
Thank you ! Indeed `git rev-list pr.Head.SHA -1 ` nails the problem !! 

I will look after [git/revlist](https://github.com/go-gitea/gitea/blob/main/modules/git/pipeline/revlist.go#L36) and will come back with a better thing, thank you for pointing this out !!!

Reply:
```suggestion
		_, err = git.NewCommand("rev-list", "-1", pr.Head.SHA).RunInDir(g.repo.RepoPath())
```

Reply:
Thanks for the suggstion, it works great and smooth ! 
I tried to get a look at [git/revlist](https://github.com/go-gitea/gitea/blob/main/modules/git/pipeline/revlist.go#L36) in the git modules, but this would need to catch the